modulus/utils/sdf.py

```diff
@@ -127,3 +128,66 @@ def signed_distance_field(
         return (sdf, sdf_hit_point_id)
     else:
         return sdf
+
+
+def sdf_to_stl(
+    field: NDArray[float],
+    threshold: float = 0.0,
+    backend: str = "warp",
+    filename: str = "output_stl.stl",
+):
+    """
+    Helper utility to create STL from input SDF using Marching Cube algorithm.
+    Wrapper around Warp's algorithm: https://nvidia.github.io/warp/modules/runtime.html#marching-cubes
+    and scikit-image's algorithm: https://scikit-image.org/docs/stable/api/skimage.measure.html#skimage.measure.marching_cubes
+
+    Parameters
+    ----------
+    field : NDArray[float]
+        SDF field array. Must be a 3D tensor of shape [nx, ny, nz].
+    threshold : float, optional
+        Target iso-surface value, by default 0.0
+    backend : str, optional
+        Backed to use. Options available warp and skimage, by default warp
+    filename : str, optional
+        Filename for output stl file, by default "output_stl.stl"
+    """
+    if backend == "warp":
+        # Convert numpy array to warp array
+        field = wp.array(field)
+
+        mc = wp.MarchingCubes(
+            field.shape[0],
+            field.shape[1],
+            field.shape[2],
+            max_verts=int(1e6),
+            max_tris=int(1e6),
+        )
+
+        # extract the surface
+        mc.surface(field=field, threshold=threshold)
+
+        # extract the vertices and faces
+        verts = mc.verts.numpy()
+        faces = mc.indices.numpy().reshape(-1, 3)
+
+    elif backend == "skimage":
```

Comment:
Is there any reason to support two backends?

Reply:
Nothing in particular, just wanted to provide multiple options - just in cases where one is better over the other. Also, this offers a non-gpu backend for someone to try.